### ISSUE TITLE
Option for disable pamltsp

### ISSUE THREAD

**Issue by kvaps** [ltsp/ltsp#81]
Hi,
We need an option for disable ltsp pam authentication.
Some users uses sssd and other methods for authentication.

Optionally we also need to keep local accounts as well. 

I would like to add two new variables:

- `LTSP_AUTH=<0|1>` - for enable or disable ltsppam module (default: 1)
- `KEEP_LOCAL_USERS=<0|1>` - for enable or disable ltsp users cleanup (default: 0)

**Comment by kvaps** [comment]
One more question, how can I include `home/*` to initramfs image without editing `/usr/share/ltsp/server/image/image.excludes` file?

**Comment by alkisg** [comment]
To override any function without introducing variables and without providing 56-xxx.sh scripts, one can use ltsp.conf parameters like the following:

```shell
[server]
PRE_IMAGE_NO_REMOVE_USERS="eval 'remove_users() { echo Omitting remove_users; }'"

[clients]
PRE_INIT_NO_PAM_MAIN="eval 'pam_main() { echo Omitting pam_main; }'"
```

To include home/* one way would be to clone image.excludes to /etc/ltsp while omitting that line. I'm not sure if mksquashfs has parameters to include files, if so, that would help.

**Comment by alkisg** [comment]
Maybe we can introduce a regex, e.g. NO_IMAGE_EXCLUDES="^(home|var)/", that would then call this:

```shell
grep -vE "$NO_IMAGE_EXCLUDES" image.excludes > $(mktemp -d)/image.excludes
```

...and then use the temp image.excludes as a parameter to mksquashfs.

**Comment by kvaps** [comment]
I like `OMIT_FUNCTIONS` way

>To include home/* one way would be to clone image.excludes to /etc/ltsp while omitting that line. 

This will not work, because both files are used, local and upstream one:
https://github.com/ltsp/ltsp/blob/master/ltsp/server/image/65-mksquashfs.sh#L30-L31

I think we have to add two more options for configure mksquashfs without creating own image.excludes file. Also consider to not use regex syntax, it might have the same  syntax:

```
NO_IMAGE_EXCLUDES="home/*"
ADD_IMAGE_EXCLUDES="
foo
bar/*
baz/ggg yyy"
```

**Comment by alkisg** [comment]
> This will not work, because both files are used

No. There are two local files: image.excludes **overrides** the upstream one, while local-image.excludes is an additional one.

I like the ADD_IMAGE_EXCLUDES and REMOVE_IMAGE_EXCLUDES approach. I think if those are implemented, we won't need local-image.excludes anymore, but it will still be nice to have the /etc/ltsp/image.excludes override in case someone wants something really customized.

Can you file a different issue or PR to discuss that part separately?
I'm not sure if we'll want to suggest multiline values by default, or if multiple variables would be better, e.g. ADD_IMAGE_EXCLUDES_FOO="foo", ADD_IMAGE_EXCLUDES_BAR="bar" etc.

And let's make this PR about implementing OMIT_FUNCTIONS.

**Comment by kvaps** [comment]
So after merging #83 and #84 local users can be enabled by specifying:
```ini
[server]
OMIT_FUNCTIONS="remove_users"
OMIT_IMAGE_EXCLUDES="home/*"
[clients]
OMIT_FUNCTIONS="pam_main"
```
